Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm

```diff
@@ -1856,16 +1874,37 @@ - (void)_performSpatialFullScreenTransition:(BOOL)enter completionHandler:(Compl
             }
 
             scene.mrui_placement.preferredChromeOptions = [originalState sceneChromeOptions];
+            if (isImageElement)
+                scene.mrui_placement.preferredChromeOptions = RSSSceneChromeOptionsNone;
 
             completionHandler();
         });
     });
 
+    auto* manager = self._manager;
+    CGFloat inWindowAlpha = 1;
+#if PLATFORM(VISION)
+    if (manager && manager->isImageElement()) {
+        if (enter) {
+            inWindowAlpha = 0;
+            NSString *appDisplayName = [[NSBundle mainBundle] objectForInfoDictionaryKey:(__bridge NSString *)_kCFBundleDisplayNameKey];
```

Comment:
Should this use `applicationVisibleName()` here? That handles the edge case where a bundle doesn't have a display name.